src/plugins/playlist/playlist.js

```diff
@@ -0,0 +1,128 @@
+import { sliceProperties } from 'utils/slicing';
+import { normalizeJsonResponse } from 'utils/api';
+import { assign } from 'utils/assign';
+import { PLAYER_EVENT } from 'utils/consts';
+import { isPlainObject } from 'utils/type-inference';
+import { extendCloudinaryConfig, getCloudinaryUrl } from 'plugins/cloudinary/common';
+
+import Playlist from './ui/playlist';
+import PlaylistWidget from './ui/playlist-widget';
+import './ui/panel/playlist-panel';
+
+const LIST_BY_TAG_PARAMS = { format: 'json', resource_type: 'video', type: 'list' };
+
+const playlist = (player, options = {}) => {
+  const _chainTarget = sliceProperties(options, 'chainTarget').chainTarget;
+  let _playlist = null;
+  let _playlistDisposer = null;
+  let _playlistWidget = null;
+
+  const disposePlaylist = () => {
+    player.removeClass('vjs-playlist');
+    const playlist = player.playlist();
+    _playlist = undefined;
+    playlist.dispose();
+    player.off('cldsourcechanged', _playlistDisposer);
+  };
+
+  const createPlaylist = (sources, options) => {
+    if (sources instanceof Playlist) {
+      _playlist = sources;
+      _playlist.resetState();
+      _playlist.currentIndex(_playlist.currentIndex());
+    } else {
+      _playlist = new Playlist(player.cloudinary, sources, options);
+      _playlist.currentIndex(0);
+    }
+
+    initPlaylistWidget();
+
+    _playlistDisposer = addSourceChangedListener();
```

Comment:
This are not new handlers, they just moved here from `playlistable`
Not sure I want to do such structural changes here, will this work the same if we have multiple players on the same page? or the same player but multiple playlists?

Reply:
@tsi because it is an old code it is considered a different scope, so feel free to merge with this code especially if you think it is risky. to my eyes though this should work exactly as it was before, but again feel free to skip now.